assets/js/payment-methods-demo/express-payment/apple-pay/normalize.js

```diff
@@ -0,0 +1,169 @@
+ */
+const normalizeLineItems = ( cartTotalItems, pending = false ) => {
+	return cartTotalItems.map( ( cartTotalItem ) => {
+		return {
+			amount: cartTotalItem.value,
+			label: cartTotalItem.label,
+			pending,
+		};
+	} );
+};
+
+/**
+ * Normalizes incoming cart shipping option items for use as shipping options
+ * with the Stripe api.
+ *
+ * @param {CartShippingOption[]}  shippingOptions An array of CartShippingOption items.
+ *
+ * @return {StripeShippingOption[]}  An array of Stripe shipping option items.
+ */
+const normalizeShippingOptions = ( shippingOptions ) => {
+	return shippingOptions.map( ( shippingOption ) => {
+		return {
+			id: shippingOption.rate_id,
+			label: shippingOption.name,
+			detail: shippingOption.description,
+			amount: parseInt( shippingOption.price, 10 ),
+		};
+	} );
+};
+
+/**
+ * Normalize shipping address information from stripe's address object to
+ * the cart shipping address object shape.
+ *
+ * @param {StripeShippingAddress} shippingAddress Stripe's shipping address item
+ *
+ * @return {CartShippingAddress} The shipping address in the shape expected by
+ * the cart.
+ */
+const normalizeShippingAddressForCheckout = ( shippingAddress ) => {
+	return {
+		first_name: shippingAddress.recipient
+			.split( ' ' )
+			.slice( 0, 1 )
+			.join( ' ' ),
+		last_name: shippingAddress.recipient
+			.split( ' ' )
+			.slice( 1 )
+			.join( ' ' ),
+		company: '',
+		address_1:
+			typeof shippingAddress.addressLine[ 0 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 0 ],
+		address_2:
+			typeof shippingAddress.addressLine[ 1 ] === 'undefined'
+				? ''
+				: shippingAddress.addressLine[ 1 ],
+		city: shippingAddress.city,
+		state: shippingAddress.region,
+		country: shippingAddress.country,
+		postcode: shippingAddress.postalCode,
+	};
+};
+
+/**
+ * Normalizes shipping option shape selection from Stripe's shipping option
+ * object to the expected shape for cart shipping option selections.
+ *
+ * @param {StripeShippingOption} shippingOption The customer's selected shipping
+ *                                              option.
+ *
+ * @return {string[]}  An array of ids (in this case will just be one)
+ */
+const normalizeShippingOptionSelectionsForCheckout = ( shippingOption ) => {
+	return [ shippingOption.id ];
+};
+
+/**
+ * Returns the billing data extracted from the stripe payment response to the
+ * CartBillingData shape.
+ *
+ * @param {StripePaymentResponse} paymentResponse Stripe's payment response
+ *                                                object.
+ *
+ * @return {CartBillingAddress} The cart billing data
+ */
+const getBillingData = ( paymentResponse ) => {
+	const source = paymentResponse.source;
+	const name = source && source.owner.name;
+	const billing = source && source.owner.address;
+	const payerEmail = paymentResponse.payerEmail || '';
+	const payerPhone = paymentResponse.payerPhone || '';
+	return {
+		first_name: name
+			? name
+					.split( ' ' )
```

Comment:
This might be common across gateways so extracting this type of thing to a util would probably be worthwhile

Reply:
The need for mapping will probably be common, but the values mapped will likely differ between gateways. Also, this is only used for the stripe apple pay gateway (because Apple Pay exposes the billing data from the customer's Apple Pay account).